[BUGFIX] Fix packages names and solve dependancy to typo3-core to its packages

sfsmfc opened the pull request:
Replace the composer names with the right one from packagist in order to avoid problem with duplicated packages.

Solve the dependancy from typo3-cms to typo3-cms-core.

Commit:
Update composer.json

restriction to typo3-core modules and cleanup composer names

Commit:
Update ext_emconf.php

update dependancies in ext_emconf.php

websi commented on ext_emconf.php:
```diff
@@ -38,13 +38,7 @@
             'realurl' => '2.3.2-2.3.99',
             'themes' => '8.7.0-8.7.99',
             'gridelements' => '8.2.0-8.2.99',
-            'solr' => '7.5.1-7.5.99',
```
Why you remove the dependencies from `ext_emconf.php`?
The dependencies from `composer.json` and `ext_emconf.php` should be the same.

sfsmfc replied:
Sorry, is fixed.

Commit:
Correct composer name for clickstorm/cs-seo

Correct composer name from clickstorm/cs-seo to Correct composer name for clickstorm/cs_seo

Commit:
Revert changes in ext_emconf from before

mabolek commented:
Tested. Looks hunky dory.
